core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/buffer_v1.go

```diff
@@ -198,17 +202,30 @@ func (b *logBuffer) dequeue(start, end int64, upkeepLimit, capacity int, upkeepS
 			remainingLogs += logsInRange
 			continue
 		}
-		if upkeepLimit > capacity {
+		if logLimit > capacity {
 			// adjust limit if it is higher than the actual capacity
-			upkeepLimit = capacity
+			logLimit = capacity
 		}
-		logs, remaining := q.dequeue(start, end, upkeepLimit)
+
+		var logs []logpoller.Log
+		remaining := 0
+
+		if minimumDequeue {
+			logs, remaining = q.dequeue(start, end, logLimit-q.dequeued[start])
```

Comment:
can this become negative? (as we override logLimit by capacity)

logically what we need here is `min(capacity, logLimit-q.dequeued[start])` where logLimit is the original logLimit

Reply:
This would only be negative when we had dequeued more logs than the log limit; and earlier in the function we have a check that compares if we have dequeued more than or as many logs as log limit, and if we have, we continue to the next iteration of the loop so this line wouldn't be reached?